### PR TITLE
php 7.4 compatibility

### PR DESCRIPTION
**Pull request by ChrisB9**
Adjusting composer.json to accept installation on php7.4 too

### CODE REVIEW

File: src/Command/DatabaseCommand.php [hunk 1]
```diff
@@ -304,6 +304,18 @@ private function generateCliParameters(DatabaseUrl $databaseUrl, bool $excludeDa
...
+        $excludeOptions = [];
+        foreach ($this->getExcludes() as $exclude) {
+            $excludeOptions[] = '--ignore-table=' . escapeshellarg($database . '.' . $exclude);
+        }
+        if ($excludeOptions !== []) {
+            return ' ' . implode(' ', $excludeOptions);
+        }
+        return '';
+    }
+
```
Comment (ChrisB9) [inline src/Command/DatabaseCommand.php:318]:
```suggestion

```

File: src/Command/DatabaseCommand.php [hunk 2]
```diff
@@ -304,6 +304,18 @@ private function generateCliParameters(DatabaseUrl $databaseUrl, bool $excludeDa
         return implode(' ', $result);
     }
 
+    private function excludeParts(string $database): string
```
Comment (Kanti) [inline src/Command/DatabaseCommand.php:307]: remove? as this is not used any ware else

File: src/Config/SshConfigGenerator.php [hunk 3]
```diff
@@ -18,8 +18,8 @@ public function findShortestPath(string $source, string $destination, SshConnect
         $shortest = [];
         $connections = $this->findAllPaths($source, $destination, $sshConnections);
         //TODO: warning found multiple Connection Possibilities, Selected one of the Shortest. these are all of them: ...
-//        if (count($connections) > 1) {
-//        }
+        //        if (count($connections) > 1) {
+        //        }
```
Comment (Kanti) [inline src/Config/SshConfigGenerator.php:22]:
```suggestion
        //if (count($connections) > 1) {
        //}
```

File: composer.json [hunk 4]
```diff
@@ -53,14 +54,18 @@
...
+    "symfony/process": "^4.0 | ^5.0",
+    "php": ">=7.2 <7.5",
     "ext-json": "*"
   },
   "require-dev": {
-    "pluswerk/grumphp-config": "2.*",
-    "bamarni/composer-bin-plugin": "^1.2"
+    "pluswerk/grumphp-config": "^3.0",
+    "bamarni/composer-bin-plugin": "^1.2",
+    "rector/rector-prefixed": "^0.7.14"
```
[the remark this answers is not in the document]
Comment (ChrisB9) [reply composer.json:65]: after the issue is being processed, since it would fail all the time

File: grumphp.yml [hunk 5]
```diff
@@ -5,3 +5,40 @@ imports:
...
+      metadata:
+        blocking: "%convention.security_checker_blocking%"
+    plus_bom_fixer:
+      metadata:
+        priority: 1
+#    phpstan:
+#      autoload_file: ~
+#      configuration: phpstan.neon
+#      level: 8
+#      force_patterns: ['/^src/']
```
Comment (ChrisB9) [inline grumphp.yml:42]: this currently checks tests too, but it should not do that (49 errors remaining)

File: src/Config/DatabaseUrl.php [hunk 6]
```diff
@@ -1,9 +1,14 @@
 <?php
 
+/** @noinspection ALL */
+
```
Comment (Kanti) [inline src/Config/DatabaseUrl.php:4]:
```suggestion
```

**Comment by Kanti** [comment]
ToDo: edit README.md

**Comment by ChrisB9** [comment]
> ToDo: edit README.md

updated

**Pull request merged**